crates/orchestrator/src/lib.rs

```diff
@@ -421,10 +425,9 @@ where
         libp2p_address: Option<Multiaddr>,
         libp2p_public_key: Option<PeerId>,
     ) -> Result<u16, ServerError> {
-        let node_index = self.latest_index;
-        self.latest_index += 1;
+        let node_index = self.nodes_identified.len();
 
-        if usize::from(node_index) >= self.config.config.num_nodes_with_stake.get() {
+        if node_index >= self.config.config.num_nodes_with_stake.get() {
```

Comment:
I think we'd want this to look up the `node_index` in the existing list and return that if it exists, so that a node restarting after `post_identity` still receives the same `node_index` it did the first time

Reply:
Hmm, the index is a u16 in this case, what list stores the indices? I see `pub_posted`, but that takes a public key, which doesn't seem to be either of the posted fields here?